### ISSUE TITLE
Shorten Training Time 

### ISSUE THREAD

**Issue by andrewcby** [microsoft/unilm#144]
I have a question - I fine-tuned the network on my own data and it took around 11 hours for 11 epoch. Although the results converged after only a few epochs, it still is very slow. So here are my two questions/suggestions:
1) Can we add something to record the intermittent accuracy? Currently in the file `eval_results.txt` there's only the accuracy of the last epoch. Would be nice to see intermittent results. And/or having an option to not display all the info like "reading" "Evaluating", etc. It would help getting the information better 🙂 
2) Is there a way to parallel the training? Having access to clusters with multiple GPUs but only able to use one seems like a waste. Thanks! 😛

**Comment by wolfshow** [comment]
@andrewcby thanks for the suggestion! We perform parallel training cross clusters during the pre-training phrase, but for most fine-tuning tasks, one GPU is enough for us.

**Comment by andrewcby** [comment]
@wolfshow, thank you for your reply! Is it possible to enable parallel training for fine tuning as well?

**Comment by wolfshow** [comment]
@andrewcby We have no bandwidth right now but will enable that once we have time.

**Comment by ranpox** [comment]
Hi @andrewcby, the model supports `DataParallel` and `DistributedDataParallel`. You can directly use multiple GPUs as usual during the fine-tuning stage.

**Comment by andrewcby** [comment]
@ranpox Hi! I ran into an issue with `DataParallel` and `DistributedDataParallel` though. I tried training on a 8 GPU cluster, and it just got stuck after loading data, at this step: 
```
05/22/2020 03:35:44 - INFO - __main__ -   Saving features into cached file ../../data/cached_train_layoutLM_128_cdip
05/22/2020 03:35:52 - INFO - __main__ -   ***** Running training *****
05/22/2020 03:35:52 - INFO - __main__ -     Num examples = 14549
05/22/2020 03:35:52 - INFO - __main__ -     Num Epochs = 4
05/22/2020 03:35:52 - INFO - __main__ -     Instantaneous batch size per GPU = 16
05/22/2020 03:35:52 - INFO - __main__ -     Total train batch size (w. parallel, distributed & accumulation) = 128
05/22/2020 03:35:52 - INFO - __main__ -     Gradient Accumulation steps = 1
05/22/2020 03:35:52 - INFO - __main__ -     Total optimization steps = 456

Epoch:   0%|          | 0/4 [00:00<?, ?it/s]
```

I have tried adjusting different models, parameters, etc. and nothing helped. Any idea why this is the case?

**Comment by ranpox** [comment]
Hi @andrewcby 
Could you please share the exact command you used? 
Actually, if you use the command in README, `DataParallel` would be enabled automatically.

**Comment by andrewcby** [comment]
Hi @ranpox, here's the command I used: 
```
python run_classification.py  --data_dir  ../../data \
                              --model_type layoutlm \
                              --model_name_or_path ../../layoutLM \
                              --output_dir /dbfs/FileStore/LayoutLMOutput3_testing_multi_GPUs \
                              --do_lower_case \
                              --max_seq_length 128 \
                              --task_name cdip \
                              --do_eval \
                              --do_train \
                              --num_train_epochs 4.0 \
                              --logging_steps 5000 \
                              --save_steps 5000 \
                              --per_gpu_train_batch_size 16 \
                              --per_gpu_eval_batch_size 16 \
                              --evaluate_during_training \
                              --fp16 \
                              --overwrite_output_dir 
```
It's basically the same with the command from README. I used the same command and trained with 1 GPU without a problem (it just took a little longer). So I switched to a multi-GPU cluster hoping it would be faster, but it got stuck. I let it run for 20 hours and it's still stuck at that step. Thanks!

**Comment by ranpox** [comment]
Duplicate of #148
